wrappers/csharp/Native.cs

```diff
@@ -86,6 +86,46 @@ public static bool VerifyPassword(byte[] password, byte[] hash)
         }
 
 #if !ANDROID && !IOS && !MAC_MODERN
+        public static long GenerateSharedKeyNative(UIntPtr nbShares, UIntPtr threshold, UIntPtr size, IntPtr[] shares)
+        {
+            if (Environment.Is64BitProcess)
+            {
+                return GenerateSharedKey64(nbShares, threshold, size, shares);
+            }
+
+            return GenerateSharedKey86(nbShares, threshold, size, shares);
+        }
+
+        public static long JoinSharesNative(UIntPtr nbShares, UIntPtr sharesLength, IntPtr[] shares, byte[] secret, UIntPtr secretLength)
+        {
+            if (Environment.Is64BitProcess)
+            {
+                return JoinShares64(nbShares, sharesLength, shares, secret, secretLength);
+            }
+
+            return JoinShares86(nbShares, sharesLength, shares, secret, secretLength);
+        }
+
+        public static long JoinSharesSizeNative(UIntPtr size)
+        {
+            if (Environment.Is64BitProcess)
+            {
+                return JoinSharesSize64(size);
+            }
+
+            return JoinSharesSize86(size);
+        }
+
+        public static long GenerateSharedKeySizeNative(UIntPtr secretLength)
+        {
+            if (Environment.Is64BitProcess)
+            {
+                return GenerateSharedKeySize64(secretLength);
+            }
+
+            return GenerateSharedKeySize86(secretLength);
+        }
+
```

Comment:
Do not expose the methods to the lib users ( Make them internal )

Reply:
Done